test/bntest.c

```diff
@@ -312,6 +313,15 @@ int main(int argc, char *argv[])
         goto err;
     (void)BIO_flush(out);
 #endif
+
+    /* print any pre-existing error on the stack */
+    ERR_print_errors_fp(stderr);
```

Comment:
@mattcaswell , just a RFC for you before merging: this line here was added because some of the previous tests don't properly clean the error queue (e.g., among many others, div by zero is tested, and the error expected, but not cleared).

Here I decided to print rather than `ERR_clear_error()` because the existing code already did so (see line 328), and because my limited PERL understanding tell me that there is some parsing of the output going on.

The reason why I raise this concern is that looking at the verbose output of this test, both on failure and on passing, `stderr` is polluted with those bogus, expected, error messages from the queue and they could be confusing.

Reply:
Yes the output is parsed and so this output is not visible to most users. You'll only get it if your request verbose output. That said I'm not sure if there is value in printing the errors really. My approval stands if you choose to clear the errors instead.

Reply:
Just pushed a fixup commit to clear the queue instead and to remove the pre-existing `ERR_print_errors_fp()` as it would be useless there after this change.

I will let this PR simmer for one day before merging, in case someone wants to chime in and comment on the proposed change.